### ISSUE TITLE
Make query processor more friendly for embedded usage

### ISSUE THREAD

**Issue by HouzuoGuo** [HouzuoGuo/tiedot#57]
Currently, the query processor assumes input to be coming from deserialized JSON, therefore some query options such as `limit` and `int-from` are expected to be of type float64, which is awkward in embedded usage scenarios.

Change the behaviour so that during query processing, int, int64, uint64 are acceptable when JSON integer is expected.

**Comment by archey** [comment]
@HouzuoGuo maybe this one can be solved now as well? So it can be fixed for good

**Comment by HouzuoGuo** [comment]
@archey Good idea, but at the moment I'm working on some features in https://github.com/HouzuoGuo/websh. Are you feeling adventurous to add those examples to example.go?
